charts/redpanda/post_install_upgrade_job.tpl.go

```diff
@@ -0,0 +1,179 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// +gotohelm:filename=_post-install-upgrade-job.go.tpl
+package redpanda
+
+import (
+	"fmt"
+	"github.com/redpanda-data/helm-charts/pkg/gotohelm/helmette"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// RedpandaEnvironmentVariables returns environment variables assigned to Redpanda
+// container.
+func RedpandaEnvironmentVariables(dot *helmette.Dot) []corev1.EnvVar {
+	values := helmette.Unwrap[Values](dot.Values)
+
+	tieredStorageConfig := values.Storage.Tiered.Config
+
+	if values.Storage.TieredConfig != nil {
+		if len(values.Storage.TieredConfig) > 0 {
+			tieredStorageConfig = values.Storage.TieredConfig
+		}
+	}
+
+	envars := []corev1.EnvVar{}
+
+	if license := GetLicense(dot); license != "" {
+		envars = append(envars, corev1.EnvVar{
+			Name:  "REDPANDA_LICENSE",
+			Value: license,
+		})
+	} else if secretReference := EnterpriseSecretNameReference(dot); secretReference != nil {
+		envars = append(envars, corev1.EnvVar{
+			Name: "REDPANDA_LICENSE",
+			ValueFrom: &corev1.EnvVarSource{
+				SecretKeyRef: secretReference,
+			},
+		})
+	}
+
+	if !IsTieredStorageEnabled(tieredStorageConfig) {
+		return envars
+	}
+
+	if !helmette.Empty(tieredStorageConfig["cloud_storage_azure_container"]) && !helmette.Empty(tieredStorageConfig["cloud_storage_azure_storage_account"]) {
```

Comment:
I think using go's builtin dict testing would be more ergonomic here?

```go
if azureContainer, ok := tieredStorageConfig["cloud_storage_azure_container"].(string); ok && azureContain != "" {
```

Reply:
Yes, but those would be 2 different configurations:
* cloud_storage_azure_container
* cloud_storage_azure_storage_account